### PR TITLE
feat(flink): implement insert dml

### PR DESCRIPTION
**Pull request by chloeh13q**
Implement INSERT DML in Flink backend.

### CODE REVIEW

File: ibis/backends/flink/__init__.py [hunk 1]
```diff
@@ -3,13 +3,15 @@
 from functools import lru_cache
 from typing import TYPE_CHECKING, Any
 
+import pandas as pd
```
Comment (gforsyth) [inline ibis/backends/flink/__init__.py:6]: I think elsewhere (like in `create_table`) we're gating this import inside of the method that needs to the `isinstance` check.  This should probably get moved down inside of `insert`

File: ibis/backends/flink/__init__.py [hunk 2]
```diff
@@ -442,3 +444,67 @@ def _get_operations(cls):
...
         return operation in cls._get_operations()
+
+    def insert(
+        self,
+        table_name: str,
+        obj: pd.DataFrame | ir.Table | list | dict,
+        database: str | None = None,
+        catalog: str | None = None,
+        overwrite: bool = False,
+    ) -> TableResult:
```
Comment (gforsyth) [inline ibis/backends/flink/__init__.py:455]: Ibis itself is already inconsistent about whether `insert` calls return anything.  The other sqlalchemy backends return `None` -- Impala and PySpark both return some `result` object. 
Is there useful information encoded in the `TableResult`?  If not, I'd say we can just execute and not return it.

File: ibis/backends/flink/__init__.py [hunk 3]
```diff
@@ -442,3 +444,67 @@ def _get_operations(cls):
...
+        """Insert data into a table.
+
+        Parameters
+        ----------
+        table_name
+            The name of the table to which data needs will be inserted
```
Comment (gforsyth) [inline ibis/backends/flink/__init__.py:461]:
```suggestion
            The name of the table to insert data into.
```

File: ibis/backends/flink/__init__.py [hunk 4]
```diff
@@ -442,3 +444,67 @@ def _get_operations(cls):
...
+        obj
+            The source data or expression to insert
```
Comment (gforsyth) [inline ibis/backends/flink/__init__.py:463]:
```suggestion
            The source data or expression to insert.
```

File: ibis/backends/flink/__init__.py [hunk 5]
```diff
@@ -442,3 +444,67 @@ def _get_operations(cls):
...
+        database
+            Name of the attached database that the table is located in.
+        catalog
+            Name of the attached catalog that the table is located in.
+        overwrite
+            If `True` then replace existing contents of table
```
Comment (gforsyth) [inline ibis/backends/flink/__init__.py:469]:
```suggestion
            If `True` then replace existing contents of table.
```

File: ibis/backends/flink/__init__.py [hunk 6]
```diff
@@ -442,3 +444,67 @@ def _get_operations(cls):
     @classmethod
     def has_operation(cls, operation: type[ops.Value]) -> bool:
         return operation in cls._get_operations()
+
+    def insert(
+        self,
+        table_name: str,
+        obj: pd.DataFrame | ir.Table | list | dict,
```
Comment (gforsyth) [inline ibis/backends/flink/__init__.py:451]: `create_table` also accepts `pyarrow.Table` -- do we also want to accept that here?

**Review by gforsyth — COMMENTED** [review]
Looks good overall -- a few questions and a few small nits.

File: ibis/backends/flink/__init__.py (hunk 2 above)
Comment (chloeh13q) [reply ibis/backends/flink/__init__.py:457]: When `TableResult` is returned, we can call `wait()` on the `TableResult` to wait for the job to finish. I think it's at least useful for debugging purposes for now, but I don't feel strongly about returning it, although `_exec_sql` also just returns the `TableResult`.

File: ibis/backends/flink/__init__.py (hunk 6 above)
Comment (chloeh13q) [reply ibis/backends/flink/__init__.py:453]: That's a good point! We definitely can - I think we can just convert pyarrow tables to pandas tables and insert them the same way.

FWIW though, the `insert()` method in alchemy and pyspark backends don't accept pyarrow tables.

Comment (chloeh13q) [reply ibis/backends/flink/__init__.py]: Added `pyarrow.Table`!

File: ibis/backends/flink/__init__.py (hunk 2 above)
Comment (gforsyth) [reply ibis/backends/flink/__init__.py:457]: That seems like a good reason to return it!

File: ibis/backends/flink/tests/test_ddl.py [hunk 7]
```diff
@@ -94,15 +113,56 @@ def test_create_table(
...
+    sink_schema = sch.Schema({"name": dt.string, "age": dt.int64, "gpa": dt.float64})
+    with tempfile.TemporaryDirectory() as tempdir:
+        con.create_table(
+            "tempdir_sink",
+            schema=sink_schema,
+            tbl_properties=tempdir_sink_configs(tempdir),
+        )
+        con.insert(
+            "tempdir_sink",
+            [("fred flintstone", 35, 1.28), ("barney rubble", 32, 2.32)],
```
Comment (gforsyth) [inline ibis/backends/flink/tests/test_ddl.py:140]: I think this should be parametrized to also cover inserting DataFrames and pyarrow tables.

**Review by gforsyth — COMMENTED** [review]
two small nits, then I think this is good to go in!

File: ibis/backends/flink/__init__.py [hunk 8]
```diff
@@ -3,11 +3,13 @@
 from functools import lru_cache
 from typing import TYPE_CHECKING, Any
 
+import pyarrow as pa
```
[the remark this answers is not in the document]
Comment (chloeh13q) [reply ibis/backends/flink/__init__.py:6]: Good catch - sorry accidentally moved it outside

**Review by gforsyth — APPROVED** [review]
:shipit:

**Pull request merged**